src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -1167,13 +1167,16 @@ private Type GetCompatibleGenericBaseClass(Type type, Type baseType)
                     out bool setterIsVirtual,
                     out bool setterIsInitOnly);
 
+                var propInfoWrapper = memberInfo as PropertyInfoWrapper;
+
                 string clrName = memberInfo.Name;
                 string runtimePropertyName = DetermineRuntimePropName(clrName, jsonPropertyName, _currentContextNamingPolicy);
                 string propertyNameVarName = DeterminePropNameIdentifier(runtimePropertyName);
 
                 return new PropertyGenerationSpec
                 {
-                    ClrName = clrName,
+                    ExactNameSpecifiedInSourceCode = propInfoWrapper?.NeedsAtSign ?? false ? "@" + memberInfo.Name : memberInfo.Name,
```

Comment:
nit:
```suggestion
                    ExactNameSpecifiedInSourceCode = propInfoWrapper?.NeedsAtSign == true ? "@" + memberInfo.Name : memberInfo.Name,
```

Reply:
Makes sense - what was I thinking?! :)

Reply:
This XNOR (or iff) pattern with three states gets tricky at times, so I memorized this syntax. 😅